### PR TITLE
PLAT-7223: Remove replication failover mode.

### PR DESCRIPTION
**Pull request by oyeliseiev-ua**
* Remove replication failover mode as not supported by SingleStore
* Disable extendeTypes by default
* Added extended type readme example

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,6 +60,44 @@ SingleStorePoolDataSource pool = new SingleStorePoolDataSource("jdbc:singlestore
...
+## Extended types support
+
+Starting with SingleStore database version 8.7.1, extended types metadata is supported. 
+This feature can be enabled using the `enable_extended_types_metadata` session variable. 
+By default, this variable is set to false for backward compatibility, meaning the system will use the standard data type format.
+
+With SingleStore JDBC driver version 1.2.6 and later, you can enable extended types metadata by setting the `enableExtendedDataTypes` parameter to true.
+When enabled, JDBC will handle extended types metadata properly.
+
+**Note: The `enableExtendedDataTypes` JDBC parameter overrides the manually set `enable_extended_types_metadata` session variable.**
```
Comment (pmishchenko-ua) [inline README.md:72]: I don't understand what does this mean... Does it refer to the global variable `enable_extended_types_metadata`, or `SET SESSION enable_extended_types_metadata = [ON|OFF]` has no effect when called from the JDBC driver?

File: src/main/java/com/singlestore/jdbc/Configuration.java [hunk 2]
```diff
@@ -53,7 +53,7 @@
  * <p>Some examples :<br>
  * {@code jdbc:singlestore://localhost:3306/database?user=greg&password=pass}<br>
  * {@code
- * jdbc:singlestore://address=(type=master)(host=master1),address=(port=3307)(type=slave)(host=slave1)/database?user=greg&password=pass}
+ * jdbc:singlestore://address=(host=master1),address=(port=3307)(host=slave1)/database?user=greg&password=pass}
```
Comment (pmishchenko-ua) [inline src/main/java/com/singlestore/jdbc/Configuration.java:56]: I think slave1 might be confusing here

File: README.md (hunk 1 above)
Comment (oyeliseiev-ua) [reply README.md:72]: means JDBC driver on connection establishment always execute query: SET SESSION enable_extended_types_metadata = true/false and will override any previously set.

Comment (pmishchenko-ua) [reply README.md]: okay, then let's replace "manually set `enable_extended_types_metadata` session variable." with "`enable_extended_types_metadata` server (global) variable"

**Review by pmishchenko-ua — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -61,6 +60,45 @@ SingleStorePoolDataSource pool = new SingleStorePoolDataSource("jdbc:singlestore
     }
 ```
 
+## Extended types support
+
+Starting with SingleStore database version 8.7.1, extended types metadata is supported.
```
Comment (adityamish) [inline README.md:65]:
```suggestion
SingleStore supports extended types metadata starting from database version 8.7.1. Enable the `enable_extended_types_metadata` session engine variable to use extended types in SingleStore. By default, support for extended types is disabled for backward compatibility and the database uses the standard data type format. 
```

File: README.md [hunk 4]
```diff
@@ -61,6 +60,45 @@ SingleStorePoolDataSource pool = new SingleStorePoolDataSource("jdbc:singlestore
     }
 ```
 
+## Extended types support
+
+Starting with SingleStore database version 8.7.1, extended types metadata is supported. 
+This feature can be enabled using the `enable_extended_types_metadata` session variable. 
+By default, this variable is set to false for backward compatibility, meaning the system will use the standard data type format.
```
Comment (adityamish) [inline README.md:67]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -61,6 +60,45 @@ SingleStorePoolDataSource pool = new SingleStorePoolDataSource("jdbc:singlestore
...
+
+With SingleStore JDBC driver version 1.2.6 and later, you can enable extended types metadata by setting the `enableExtendedDataTypes` parameter to true.
+Based on this parameter, the JDBC driver automatically sets the `enable_extended_types_metadata` session variable during connection initialization.
```
Comment (adityamish) [inline README.md:70]:
```suggestion
**Note**: Based on the value of the `enableExtendedDataTypes` parameter, the driver automatically configures the value of `enable_extended_types_metadata` engine variable while initializing the connection, which overrides the current value of this engine variable.
```

File: README.md [hunk 6]
```diff
@@ -61,6 +60,45 @@ SingleStorePoolDataSource pool = new SingleStorePoolDataSource("jdbc:singlestore
...
 
+## Extended types support
+
+Starting with SingleStore database version 8.7.1, extended types metadata is supported. 
+This feature can be enabled using the `enable_extended_types_metadata` session variable. 
+By default, this variable is set to false for backward compatibility, meaning the system will use the standard data type format.
+
+With SingleStore JDBC driver version 1.2.6 and later, you can enable extended types metadata by setting the `enableExtendedDataTypes` parameter to true.
+Based on this parameter, the JDBC driver automatically sets the `enable_extended_types_metadata` session variable during connection initialization. 
+When enabled, the JDBC driver will properly handle extended types metadata.
```
Comment (adityamish) [inline README.md:71]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -61,6 +60,45 @@ SingleStorePoolDataSource pool = new SingleStorePoolDataSource("jdbc:singlestore
...
+
+**Note: The JDBC driver always sets the `enable_extended_types_metadata` session variable to false or true during connection initialization, which may override any previously set value. By default, `enableExtendedDataTypes` is set to false.**
```
Comment (adityamish) [inline README.md:73]:
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -61,6 +60,45 @@ SingleStorePoolDataSource pool = new SingleStorePoolDataSource("jdbc:singlestore
...
+
+Extended types:
+
+| Type                                                                                         | Affected changes                                                                                     |
```
Comment (adityamish) [inline README.md:77]:
```suggestion
| Data Type                                                                                         | Updates                                                                                     |
```

File: README.md [hunk 9]
```diff
@@ -61,6 +60,45 @@ SingleStorePoolDataSource pool = new SingleStorePoolDataSource("jdbc:singlestore
...
+|----------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------|
+| [Vector](https://docs.singlestore.com/cloud/reference/sql-reference/data-types/vector-type/) | Added `com.singlestore.jdbc.type.Vector` type to read/write Vector data<br/>Updated metadata methods |
```
Comment (adityamish) [inline README.md:79]:
```suggestion
| [Vector](https://docs.singlestore.com/cloud/reference/sql-reference/data-types/vector-type/) | Added `com.singlestore.jdbc.type.Vector` type to read/write Vector data.<br/>Updated metadata methods. |
```

File: README.md [hunk 10]
```diff
@@ -61,6 +60,45 @@ SingleStorePoolDataSource pool = new SingleStorePoolDataSource("jdbc:singlestore
...
+| [BSON](https://docs.singlestore.com/cloud/reference/sql-reference/data-types/bson-type/)     | Updated metadata methods                                                                             |
```
Comment (adityamish) [inline README.md:80]:
```suggestion
| [BSON](https://docs.singlestore.com/cloud/reference/sql-reference/data-types/bson-type/)     | Updated metadata methods.                                                                             |
```

File: src/main/resources/driver.properties [hunk 11]
```diff
@@ -70,5 +70,5 @@ consoleLogFilepath=Console log file path name. Default is null. If this paramete
 printStackTrace=If set, prints stack trace for TRACE log level. Default is false.
 maxPrintStackSizeToLog=Limit number of stack trace lines to log. Default is 10.
 disablePipeline=option permitting disabling pipelining
-enableExtendedDataTypes=If set, this enables extended data types by setting the enable_extended_types_metadata engine variable, allowing the JDBC API to work with extended types such as Vector, BSON, etc. This feature has been available since SingleStore v8.7.1. The default value is true.
+enableExtendedDataTypes=If set, this enables extended data types by setting the enable_extended_types_metadata engine variable, allowing the JDBC API to work with extended types such as Vector, BSON, etc. This feature has been available since SingleStore v8.7.1. The default value is false.
```
Comment (adityamish) [inline src/main/resources/driver.properties:73]:
```suggestion
enableExtendedDataTypes=If set to `true`, enables extended data types by enabling the `enable_extended_types_metadata` engine variable, allowing the JDBC API to work with extended types such as Vector, BSON, etc. Extended types are supported in SingleStore database version 8.7.1. and later. The default value is `false`.
```

**Review by adityamish — APPROVED** [review]

**Pull request merged**